packages/core/src/hl7.ts

```diff
@@ -160,10 +192,96 @@ export class Hl7Message {
         ],
         this.context
       ),
-      new Hl7Segment(['MSA', 'AA', controlId, 'OK'], this.context),
+      new Hl7Segment(['MSA', ackCode, controlId, TEXT_MSG_FOR_ACK_CODE[ackCode]], this.context),
+      ...(options?.errSegment
+        ? [options?.errSegment instanceof Hl7Segment ? options.errSegment : this.buildDefaultErrSegment(options)]
+        : []),
     ]);
   }
 
+  private buildDefaultErrSegment(options?: Hl7AckOptions): Hl7Segment {
```

Comment:
Hey, nice work on this. Good call on adding the `AE` and `CE` support, that's a solid improvement.

On the error-building helpers: I'm thinking we should avoid baking business logic (like default error messages) into the library itself. The library should provide the tools, but the calling application should decide what the error content is.

What do you think about changing the API to just accept a pre-built `HL7Segment` for the error part? That would give the user full control and keep our library's responsibility clean.

Also, I'd prefer to remove the optional "Information" segment from the standard "AA" to keep the success path minimal.

Happy to chat through it if you'd like.

Reply:
No I think that makes sense, removed the default ERR segment and made it so that if you want an errSegment you have to pass it in 👍